clang/lib/Analysis/CallGraph.cpp

```diff
@@ -136,6 +137,37 @@ class CGBuilder : public StmtVisitor<CGBuilder> {
     }
   }
 
+  void VisitIfStmt(IfStmt *If) {
+    if (G->shouldSkipConstantExpressions()) {
+      if (llvm::Optional<Stmt *> ActiveStmt =
+              If->getNondiscardedCase(G->getASTContext())) {
+        if (*ActiveStmt)
```

Comment:
Under what circumstances can we get a valid `Optional` that wraps a null pointer?

Reply:
Heh... the logic to `getNondiscardedCase` is... interesting.  If this _IS_ a constexpr-if, we get a valid optional.  If the non-discarded statement doesn't exist[0], we get a nullptr in the optional.

[0] in the case where it IS a constexpr-if, but the 'else' would be the non-discarded statement but there _IS_ no else.

Reply:
Ahhh, okay! Thank you for not making me track that down myself. :-D